### PR TITLE
Ci: Release build for CI 

### LINKED ISSUE #84: Fix realease APK

### PR DESCRIPTION
**Pull request by RemIsMyWaifuu**
Modified .apk workflow to build a release of the application and prints SHA-1 fingerprint.

Fingerprint is needed for Google Sign-in, and proguard-rules.pro and build.gradle.kts minimize the size of the resulting apk.
One needs to manually upload the sha-1 digest to Firestore for Google Sign-in to work.

Other release options such as performance optimizations and size reduction were also added. 

Closes #84

### CODE REVIEW

File: .github/workflows/apk.yml [hunk 1]
```diff
@@ -1,16 +1,16 @@
-name: Android Build
+name: Android Release Build
 
 on:
   workflow_dispatch:
     inputs:
       name:
         description: "Ex: milestone name"
-        default: "Mn"
+        default: "TestRelease"
```
Comment (appellet) [inline .github/workflows/apk.yml:8]: You must give it a name that is not a “test”, as it will be merged with the main file.

File: .github/workflows/apk.yml [hunk 2]
```diff
@@ -23,39 +23,43 @@ jobs:
       - name: Checkout code
         uses: actions/checkout@v3
 
-        # Load google-services.json and local.properties from the secrets
-      - name: Decode secrets
+      - name: Set up JDK
+        uses: actions/setup-java@v3
+        with:
+          distribution: 'zulu'
+          java-version: '17'
```
Comment (appellet) [inline .github/workflows/apk.yml:30]: I think it's best to stay consistent with our `build.gradle.kts`, however I'm not sure if it's the same as the version specified in the image below :

image

File: .github/workflows/apk.yml [hunk 3]
```diff
@@ -64,26 +68,27 @@ jobs:
...
         id: create_release
         uses: actions/create-release@v1
         env:
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN}}
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
         with:
-          tag_name: ${{ github.run_number}}
-          release_name: ${{ github.event.inputs.name }} (#${{ github.run_number}})
+          tag_name: ${{ github.run_number }}
+          release_name: ${{ github.event.inputs.name }} (#${{ github.run_number }})
```
Comment (appellet) [inline .github/workflows/apk.yml:81]: Are the space really necessary or is it a mistake ?

File: app/proguard-rules.pro
Comment (appellet) [inline app/proguard-rules.pro:1]: Could you explain where this file come from ?

**Review by appellet — CHANGES_REQUESTED** [review]
This PR is usefull for creating a release APK, but a few clarifications are needed to approve. 
In general, it could only be useful for our project :+1:

File: .github/workflows/apk.yml (hunk 3 above)
Comment (RemIsMyWaifuu) [reply .github/workflows/apk.yml:81]: It looks nicer that way

File: app/proguard-rules.pro
Comment (RemIsMyWaifuu) [reply app/proguard-rules.pro:1]: It comes from the sample project. It's a file that defines optimization settings for R8 when you build an APK in release mode. It removes unnecessary files and the like.

File: .github/workflows/apk.yml (hunk 2 above)
Comment (RemIsMyWaifuu) [reply .github/workflows/apk.yml:30]: According to my own research, as long as we've got the sourceCompatibility and targetCompatibility set to 11, it will only compile to Java 11 language features. Keeping it to 17 is purely arbitrary and because of the previous workflow definition. I see no need to change

File: .github/workflows/apk.yml (hunk 1 above)
Comment (RemIsMyWaifuu) [reply .github/workflows/apk.yml:8]: Thank you for catching this, I will change it.

**Comment by RemIsMyWaifuu** [comment]
All pertinent request changes were made, CI pipeline works again. We now have the fingerprint of the signing key, and we can get working builds that support Google sign-in.

TLDR: Green checks => all good time to approve the PR

**Review by appellet — APPROVED** [review]
Requested changes and points requiring clarification have been resolved.
Everything looks shipshape and ready to sail into the main branch! 🛳️🚀
